test/units/EverlongERC4626.t.sol

```diff
@@ -0,0 +1,254 @@
+        vm.stopPrank();
+
+        // 2. Confirm that Everlong's balance is less than Hyperdrive's
+        //    minimum transaction amount.
+        assertLt(
+            ERC20Mintable(everlong.asset()).balanceOf(address(everlong)),
+            hyperdrive.getPoolConfig().minimumTransactionAmount,
+            "Everlong balance should be below min tx amount after single depositor deposit+rebalance"
+        );
+
+        // 3. Confirm that Bob's share balance equals the total supply of shares.
+        assertEq(
+            everlong.balanceOf(bob),
+            everlong.totalSupply(),
+            "single depositor share balance should equal total supply of shares"
+        );
+
+        // 4. Confirm `shares_issued = assets_deposited * (10 ^ decimalsOffset)`
+        assertEq(
+            everlong.balanceOf(bob),
+            depositAmount * (10 ** everlong.decimalsOffset())
+        );
+    }
+
+    /// @dev Validates that `redeem()` will close the necessary positions with a
+    ///       mature position sufficient to cover withdrawal amount.
+    function test_redeem_close_positions_mature() external {
+        // Initialize the Hyperdrive instance by adding liquidity from Alice.
+        uint256 fixedRate = 0.5e18;
+        uint256 contribution = 10_000e18;
+        initialize(alice, fixedRate, contribution);
+
+        // Deposit assets into Everlong as Bob.
+        mintApproveEverlongBaseAsset(bob, 10e18);
+        vm.startPrank(bob);
+        everlong.deposit(1e18, bob);
+        vm.stopPrank();
+
+        // Advance time by a checkpoint so that future deposits result in
+        // new positions.
+        advanceTime(hyperdrive.getPoolConfig().checkpointDuration, 0);
+
+        // Confirm that Everlong currently does not have a matured position.
+        assertFalse(everlong.hasMaturedPositions());
+
+        // Deposit assets into Everlong as Celine to create another position.
+        mintApproveEverlongBaseAsset(celine, 2e18);
+        vm.startPrank(celine);
+        everlong.deposit(2e18, celine);
+        vm.stopPrank();
+
+        // Confirm that Everlong now has two positions.
+        assertEq(everlong.getPositionCount(), 2);
+
+        // Advance time to mature Bob's position but not Celine's.
+        advanceTime(everlong.getPosition(0).maturityTime - block.timestamp, 0);
+
+        // Confirm that Everlong now has a matured position.
+        assertTrue(everlong.hasMaturedPositions());
+
+        // Redeem all of Bob's shares.
+        vm.startPrank(bob);
+        everlong.redeem(everlong.balanceOf(bob), bob, bob);
+        vm.stopPrank();
+
+        // Confirm that Everlong has two immature positions:
+        // 1. Created from Celine's deposit.
+        // 2. Created from rebalance on Bob's redemption.
+        assertEq(everlong.getPositionCount(), 2);
+        assertGt(everlong.getPosition(0).maturityTime, block.timestamp);
+    }
+
+    /// @dev Validates that `redeem()` will close all positions when closing
```

Comment:
Again, I remember talking about not doing this to reduce complexity/increase security.